eng/pipelines/templates/BuildAndTest.yml

```diff
@@ -86,29 +87,30 @@ steps:
         HelixBuild: $(Build.BuildNumber)
         HelixAccessToken: $(HelixApiAccessToken)
 
-    - task: CopyFiles@2
-      inputs:
-        Contents: '${{ parameters.repoArtifactsPath }}/**/*.cobertura.xml'
-        TargetFolder: '${{ parameters.repoArtifactsPath }}/CodeCoverage'
-        flattenFolders: true
-      displayName: Gather code coverage results
+  - task: CopyFiles@2
+    inputs:
+      Contents: '${{ parameters.repoArtifactsPath }}/**/*.cobertura.xml'
+      TargetFolder: '${{ parameters.repoArtifactsPath }}/CodeCoverage'
+      flattenFolders: true
+    displayName: Gather code coverage results
 
-    - ${{ if eq(parameters.runAsPublic, 'true') }}:
-      - task: PublishPipelineArtifact@1
-        displayName: Publish coverage results (cobertura.xml)
-        inputs:
-          targetPath: '${{ parameters.repoArtifactsPath }}/CodeCoverage'
-          artifactName: '$(Agent.JobName)_CodeCoverageResults'
-          publishLocation: 'pipeline'
+  - ${{ if eq(parameters.runAsPublic, 'true') }}:
+    - task: PublishPipelineArtifact@1
+      displayName: Publish coverage results (cobertura.xml)
+      inputs:
+        targetPath: '${{ parameters.repoArtifactsPath }}/CodeCoverage'
+        artifactName: '$(Agent.JobName)_CodeCoverageResults'
+        publishLocation: 'pipeline'
 
-    - ${{ if ne(parameters.runAsPublic, 'true') }}:
-      - task: 1ES.PublishPipelineArtifact@1
-        displayName: Publish code coverage results
-        inputs:
-          targetPath: '${{ parameters.repoArtifactsPath }}/CodeCoverage'
-          artifactName: '$(Agent.JobName)_CodeCoverageResults'
+  - ${{ if ne(parameters.runAsPublic, 'true') }}:
+    - task: 1ES.PublishPipelineArtifact@1
+      displayName: Publish code coverage results
+      inputs:
+        targetPath: '${{ parameters.repoArtifactsPath }}/CodeCoverage'
+        artifactName: '$(Agent.JobName)_CodeCoverageResults'
 
-  - ${{ if eq(parameters.isWindows, 'true') }}:
+  # Run on windows, for internal pipeline, or public+non-helix-tests job
+  - ${{ if and(eq(parameters.isWindows, 'true'), or(ne(parameters.runAsPublic, 'true'), ne(parameters.runHelixTests, 'true'))) }}:
```

Comment:
Is there a way to make this a bit more readable? It is really hard to parse/understand for me.

Reply:
```yml
  # Run on windows, for internal pipeline, or public+non-helix-tests job
  - ? ${{ if and(
               eq(parameters.isWindows, 'true'),
               or(
                  ne(parameters.runAsPublic, 'true'),
                  ne(parameters.runHelixTests, 'true'))) }}
    : # on a new line as the condition above is flattened into a single line
    - script: ${{ parameters.buildScript }}
              -pack
              -sign $(_SignArgs)
              -publish $(_PublishArgs)
              -configuration ${{ parameters.buildConfig }}
              /bl:${{ parameters.repoLogPath }}/pack.binlog
              /p:Restore=false /p:Build=false
              $(_OfficialBuildIdArgs)
      displayName: Pack, Sign, and Publish
```

I'm not sure how much that helps readability. I can't add comments inline because that would be part of the expression then. I could add a variables section to name parts of this like `isInternalOrNonHelixTestsJob`?

Reply:
Merging the PR, but I'll open a new one for any suggested changes from this.